RFS/src/test/java/com/rfs/cms/WorkCoordinatorTest.java

```diff
@@ -0,0 +1,179 @@
+//            return objMapper.readTree(r.getEntity().getContent());
+//        });
+//        Assertions.assertEquals("updated", response4.get("result").textValue());
+//        Assertions.assertTrue(doc4.get("_source").get("completedAt").longValue() > 0);
+//        log.info("doc4="+doc4);
+    }
+
+    //@Test
+    public void testAcquireLeaseForQuery() throws Exception {
+        var objMapper = new ObjectMapper();
+        final var NUM_DOCS = 40;
+        try (var workCoordinator = new OpenSearchWorkCoordinator(httpClientSupplier.get(),
+                3600, "docCreatorWorker")) {
+            for (var i = 0; i < NUM_DOCS; ++i) {
+                final var docId = "R" + i;
+                workCoordinator.createUnassignedWorkItem(docId);
+            }
+        }
+
+        for (int runs=0; runs<2; ++runs) {
+            final var seenWorkerItems = new ConcurrentHashMap<String, String>();
+            var allFutures = new ArrayList<CompletableFuture<Void>>();
+            final var expiration = Duration.ofSeconds(5);
+            for (int i = 0; i < NUM_DOCS; ++i) {
+                int finalI = i;
+                int finalRuns = runs;
+                var cf = CompletableFuture.supplyAsync(() ->
+                        getWorkItemAndVerity(finalRuns + "-" + finalI, seenWorkerItems, expiration));
+                allFutures.add(cf);
+            }
+            CompletableFuture.allOf(allFutures.toArray(CompletableFuture[]::new)).join();
+            Assertions.assertEquals(NUM_DOCS, seenWorkerItems.size());
+
+            try (var workCoordinator = new OpenSearchWorkCoordinator(httpClientSupplier.get(),
+                    3600, "firstPass_NONE")) {
+                var nextWorkItem = workCoordinator.acquireNextWorkItem(Duration.ofSeconds(2));
+                log.error("Next work item picked=" + nextWorkItem);
+                Assertions.assertNull(nextWorkItem);
+            }
+
+            Thread.sleep(expiration.multipliedBy(2).toMillis());
+        }
+    }
+
+    static AtomicInteger nonce = new AtomicInteger();
+    @SneakyThrows
+    private static Void getWorkItemAndVerity(String workerSuffix, ConcurrentHashMap<String, String> seenWorkerItems,
+                                             Duration expirationWindow) {
+        try (var workCoordinator = new OpenSearchWorkCoordinator(httpClientSupplier.get(),
+                3600, "firstPass_"+ workerSuffix)) {
+            var doneId = DUMMY_FINISHED_DOC_ID + "_" + nonce.incrementAndGet();
+            workCoordinator.createOrUpdateLeaseForDocument(doneId, 1);
+            workCoordinator.completeWorkItem(doneId);
+
+            var nextWorkItem = workCoordinator.acquireNextWorkItem(expirationWindow);
+            log.error("Next work item picked=" + nextWorkItem);
+            Assertions.assertNotNull(nextWorkItem);
+            Assertions.assertNotNull(nextWorkItem.workItemId);
+            Assertions.assertTrue(nextWorkItem.leaseExpirationTime.isAfter(Instant.now()));
```

Comment:
Doesn't seem to make sense to compare 'local' time against the clusters timestamp, that seems like a source of random failure in the future

Reply:
Right now, the clients (Document migration workers) handle leases by the 'honor system'.  If the clock on the client is way off, we'll have multiple parties stepping on each others toes.  This is a simple sanity check to build some protection against things from completely falling apart.

I'd like to maintain the client time for the kill-switch based off of the cluster time, as it's returned in the http headers of each target request.  The responses may be for different nodes and there could be skew there, so that could be problematic too.

Reply:
There is the general problem of a single source of truth for time I think we don't need to verify that scenario in this case of these tests.  What would we do about a failure at this line - it seems like unactionable failure to me?

I'd rather remove the validation of this concept from these tests and bake them into separate more focused test cases that assure the time management/coordination is working as expected - this seems like a good candidate for a follow up (and I don't think is a code requirement for the moment).  What do you think?